src/data_structures/src/array_ext.cairo

```diff
@@ -135,6 +136,43 @@ impl ArrayImpl<T, impl TCopy: Copy<T>, impl TDrop: Drop<T>> of ArrayTraitExt<T>
     ) -> Option<usize> {
         self.span().index_of_max()
     }
+
+    fn dedup<impl TPartialEq: PartialEq<T>>(mut self: Array<T>) -> Array<T> {
+        let len = self.len();
+
+        if(len <= 1) {
+            return self;
+        }
+
+        let mut ret = array![];
+
+        let mut last_value = match self.pop_front() {
+            Option::Some(v) => v,
+            Option::None => {
+                return ret;
+            },
+        };
```

Comment:
as you already made sure len >1 you can just pop_front().unwrap()

Reply:
Done